refactor: Deleted unused hooks, removed unused exports and imports

Anush008 opened the pull request:
<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.
  
  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.
  
  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the Open Sauced Contributing Guide: https://github.com/open-sauced/open-sauced/blob/HEAD/CONTRIBUTING.md#create-a-pull-request.
  - 📖 Read the Open Sauced Code of Conduct: https://github.com/open-sauced/open-sauced/blob/HEAD/CODE_OF_CONDUCT.md.
  - 👷‍♀️ Create small PRs. In most cases, this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This pull request aims to improve the codebase through a series of refactorings, which include removing unused interface exports, removing multiple instances of unused imports and deleting redundant hooks as reported in issue #1025 

## Related Tickets & Documents
Closes #1025 

## Mobile & Desktop Screenshots/Recordings
No visual changes.


## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## Note

* The [repository-table component](https://github.com/open-sauced/insights/blob/beta/components/molecules/RepositoryTable/repository-table.tsx) mentioned in the issue #1025 is being imported and used at [repository-table.stories.tsx](https://github.com/open-sauced/insights/blob/3ba6253a4876e9209e42607d8e109e6a9293b20d/stories/molecules/repository-table.stories.tsx). So it hasn't been removed.

brandonroberts commented:
Thanks @Anush008. The [repositories-table.stories.tsx](https://github.com/open-sauced/insights/blob/beta/stories/molecules/repositories-table.stories.tsx) isn't being used in the application, so it and its associated stories can be deleted

Anush008 commented:
Hey @brandonroberts. It is actually [repository-table.stories.tsx](https://github.com/open-sauced/insights/blob/3ba6253a4876e9209e42607d8e109e6a9293b20d/stories/molecules/repository-table.stories.tsx) and not [repositories-table.stories.tsx](https://github.com/open-sauced/insights/blob/3ba6253a4876e9209e42607d8e109e6a9293b20d/stories/molecules/repositories-table.stories.tsx), which was using the the RepoSelectTable and transitively the RepositoryRow components. I've corrected the note in the PR description.
Also, as you said these components aren't being using in the application so I've removed them and the story.
